### PR TITLE
Updatetype/ct sc/username

### PR DESCRIPTION
**Pull request by toppy007**
Please review

added a username to the signup form
added mongoDB instance for the username
added unit test in user.js 
altered unit test to compensate for new instance.

fixed bug in tokens test async and awaits beforALL().

ran jest & cypress. passed all test on second run (front and back).

might be a bug if not passing first time? discuss

### CODE REVIEW

File: api/controllers/tokens.js [hunk 1]
```diff
@@ -9,6 +9,7 @@ const SessionsController = {
 
     User.findOne({ email: email }).then(async (user) => {
       if (!user) {
+        console.log(user)
```
Comment (siqbal181) [inline api/controllers/tokens.js]: Probably can remove the log here unless it's needed for something else?

File: api/spec/controllers/tokens.spec.js [hunk 2]
```diff
@@ -16,7 +17,7 @@ describe("/tokens", () => {
   test("a token is returned when creds are valid", async () => {
     let response = await request(app)
       .post("/tokens")
-      .send({email: "test@test.com", password: "12345678"})
+      .send({email: "test@test.com", password: "12345678" })
```
Comment (siqbal181) [inline api/spec/controllers/tokens.spec.js]: Just a q, is this test ok without the username?

File: api/spec/models/user.spec.js [hunk 3]
```diff
@@ -49,9 +52,19 @@ describe("User model", () => {
...
           email: "someone@example.com",
           password: "password",
+          username: "username",
         });
         done();
       });
     });
   });
+
+  it("can make a username ", () => {
```
Comment (siqbal181) [inline api/spec/models/user.spec.js]: Just a q if this test is something relating to the model or if it belongs more in the controller which creates the user?

File: frontend/src/components/auth/LoginForm.js [hunk 4]
```diff
@@ -12,7 +12,7 @@ const LogInForm = ({ navigate }) => {
       headers: {
         'Content-Type': 'application/json',
       },
-      body: JSON.stringify({ email: email, password: password })
+      body: JSON.stringify({ email: email, password: password, })
```
Comment (siqbal181) [inline frontend/src/components/auth/LoginForm.js]: Just a q.- is this just excluding username as this is just what the user puts for login?

File: frontend/src/components/user/SignUpForm.js [hunk 5]
```diff
@@ -32,11 +33,16 @@ const SignUpForm = ({ navigate }) => {
     setPassword(event.target.value)
   }
 
+  const handleUsernameChange = (event) => {
```
Comment (siqbal181) [inline frontend/src/components/user/SignUpForm.js:36]: Again, just a q for my understanding! Why did we need to make a new const for handleUserName change, on line 33 the setPassword is just done within the handleSubmit above. Sorry if its obvious!

**Review by siqbal181 — CHANGES_REQUESTED** [review]
Awesome work team! Just some questions (mainly might be because I am unsure!) but looks like some great work)

File: api/controllers/tokens.js (hunk 1 above)
Comment (toppy007) [reply api/controllers/tokens.js]: Yeah will do! was just using it to debug

File: api/spec/models/user.spec.js (hunk 3 above)
Comment (toppy007) [reply api/spec/models/user.spec.js]: cool yeah will rename that to make it more readable

File: frontend/src/components/auth/LoginForm.js (hunk 4 above)
Comment (toppy007) [reply frontend/src/components/auth/LoginForm.js]: yes no username required for login I guess

File: frontend/src/components/user/SignUpForm.js (hunk 5 above)
Comment (toppy007) [reply frontend/src/components/user/SignUpForm.js:36]: Think the handleUsernameChange function is a callback function that is called whenever the value of the Username input field changes in the sign-up form.

but i'm not totally 100percent about it

**Review by toppy007 — COMMENTED** [review]
have made changes suggested

File: api/spec/controllers/tokens.spec.js (hunk 2 above)
Comment (toppy007) [reply api/spec/controllers/tokens.spec.js]: have changed this to include changes, was a test meant to fail so was not required to some extent

**Review by siqbal181 — APPROVED** [review]

**Pull request merged**